cloud/deploy/deploy.go

```diff
@@ -148,6 +148,73 @@ func deployDags(path, runtimeID string, client astro.Client) error {
 	return nil
 }
 
+func deployImage(c *config.Context, deployInput InputDeploy, deployInfo deploymentInfo, client astro.Client) error { // nolint
```

Comment:
consider changing `deployInput`'s type to `*InputDeploy` and remove `// nolint` as it is more efficient to pass it by reference.

Reply:
I tried doing this, it require a lot of changes at a variety of places and updating a bunch of test cases. Happy to handle it in a separate PR for easy review

Reply:
made this change so resolving it.